File: backend/src/main/java/io/apptales/minipos/service/PosService.java
package io.apptales.minipos.service;

import io.apptales.minipos.data.StockTransactionType;
import io.apptales.minipos.data.model.Product;
import io.apptales.minipos.domain.dto.ProductDto;

import java.util.List;

public interface PosService {
    Product addProduct(String addedBy, ProductDto productDto);
    List<Product> addProducts(String addedBy, List<ProductDto> productDtos);
    List<Product> getAllProducts();
    List<Product> searchProductsWithName(String name);
    Product searchProductWithProductId(String productId);
    Product addProduct(ProductDto dto);
    Product updateProduct(ProductDto dto);
    Product updateProductStock(String productId, Long newStock, StockTransactionType stockTransactionType);
    void deleteProduct(String productId);
}

File: android/app/src/main/java/io/kdbrian/minipos/android/data/remote/repo/PosService.kt
package io.kdbrian.minipos.android.data.remote.repo

import src.main.graphql.AddProductMutation
import src.main.graphql.GetAllProductsQuery
import src.main.graphql.GetProductWithIdQuery
import src.main.graphql.SearchProductsWithNameQuery
import src.main.graphql.UpdateProductStockMutation
import src.main.graphql.type.ProductDto
import src.main.graphql.type.StockTransactionType

interface PosService {

    suspend fun addProduct(product: ProductDto): Result<AddProductMutation.Data>
    suspend fun getAllProducts(): Result<GetAllProductsQuery.Data>
    suspend fun getProductWithId(productId: String): Result<GetProductWithIdQuery.Data>
    suspend fun searchProductsWithName(name: String): Result<SearchProductsWithNameQuery.Data>
    suspend fun updateProductStock(
        productId: String,
        newStock: Int,
        transactiontype: StockTransactionType
    ): Result<UpdateProductStockMutation.Data>



}